File: MiniDOMBase/DMD/Anagrafica/Banche/ContiCorrente/ContoCorrenteCursor.vb
Imports minidom
Imports minidom.Sistema
Imports minidom.Databases



Partial Public Class Anagrafica



    Public Class ContoCorrenteCursor
        Inherits DBObjectCursor(Of ContoCorrente)

        Private m_Nome As New CCursorFieldObj(Of String)("Nome")
        Private m_Numero As New CCursorFieldObj(Of String)("Numero")
        Private m_IBAN As New CCursorFieldObj(Of String)("IBAN")
        Private m_SWIFT As New CCursorFieldObj(Of String)("SWIFT")
        Private m_IDBanca As New CCursorField(Of Integer)("IDBanca")
        Private m_NomeBanca As New CCursorFieldObj(Of String)("NomeBanca")
        Private m_DataApertura As New CCursorField(Of Date)("DataApertura")
        Private m_DataChiusura As New CCursorField(Of Date)("DataChiusura")
        Private m_Saldo As New CCursorField(Of Decimal)("Saldo")
        Private m_SaldoDisponibile As New CCursorField(Of Decimal)("SaldoDisponibile")
        Private m_StatoContoCorrente As New CCursorField(Of StatoContoCorrente)("StatoContoCorrente")
        Private m_Flags As New CCursorField(Of ContoCorrenteFlags)("Flags")

        Public Sub New()
        End Sub

        Public ReadOnly Property Nome As CCursorFieldObj(Of String)
            Get
                Return Me.m_Nome
            End Get
        End Property

        Public ReadOnly Property Numero As CCursorFieldObj(Of String)
            Get
                Return Me.m_Numero
            End Get
        End Property

        Public ReadOnly Property IBAN As CCursorFieldObj(Of String)
            Get
                Return Me.m_IBAN
            End Get
        End Property

        Public ReadOnly Property SWIFT As CCursorFieldObj(Of String)
            Get
                Return Me.m_SWIFT
            End Get
        End Property

        Public ReadOnly Property IDBanca As CCursorField(Of Integer)
            Get
                Return Me.m_IDBanca
            End Get
        End Property

        Public ReadOnly Property NomeBanca As CCursorFieldObj(Of String)
            Get
                Return Me.m_NomeBanca
            End Get
        End Property

        Public ReadOnly Property DataApertura As CCursorField(Of Date)
            Get
                Return Me.m_DataApertura
            End Get
        End Property

        Public ReadOnly Property DataChiusura As CCursorField(Of Date)
            Get
                Return Me.m_DataChiusura
            End Get
        End Property

        Public ReadOnly Property Saldo As CCursorField(Of Decimal)
            Get
                Return Me.m_Saldo
            End Get
        End Property

        Public ReadOnly Property SaldoDisponibile As CCursorField(Of Decimal)
            Get
                Return Me.m_SaldoDisponibile
            End Get
        End Property

        Public ReadOnly Property StatoContoCorrente As CCursorField(Of StatoContoCorrente)
            Get
                Return Me.m_StatoContoCorrente
            End Get
        End Property

        Public ReadOnly Property Flags As CCursorField(Of ContoCorrenteFlags)
            Get
                Return Me.m_Flags
            End Get
        End Property

        Protected Friend Overrides Function GetConnection() As CDBConnection
            Return APPConn
        End Function

        Protected Overrides Function GetModule() As CModule
            Return Anagrafica.ContiCorrente.Module
        End Function

        Public Overrides Function GetTableName() As String
            Return "tbl_ContiCorrenti"
        End Function
    End Class




End Class

File: MiniDOMBase/DMD/Anagrafica/Banche/ContiCorrente/ContoCorrenteCursor.cs
using System;
using DMD;
using DMD.XML;
using DMD.Databases;
using minidom;
using static minidom.Sistema;

namespace minidom
{
    public partial class Anagrafica
    {

        /// <summary>
        /// Cursore sulla tabella dei conti corrente
        /// </summary>
        [Serializable]
        public class ContoCorrenteCursor 
            : minidom.Databases.DBObjectCursor<ContoCorrente>
        {
            private DBCursorStringField m_Nome = new DBCursorStringField("Nome");
            private DBCursorStringField m_Numero = new DBCursorStringField("Numero");
            private DBCursorStringField m_IBAN = new DBCursorStringField("IBAN");
            private DBCursorStringField m_SWIFT = new DBCursorStringField("SWIFT");
            private DBCursorField<int> m_IDBanca = new DBCursorField<int>("IDBanca");
            private DBCursorStringField m_NomeBanca = new DBCursorStringField("NomeBanca");
            private DBCursorField<DateTime> m_DataApertura = new DBCursorField<DateTime>("DataApertura");
            private DBCursorField<DateTime> m_DataChiusura = new DBCursorField<DateTime>("DataChiusura");
            private DBCursorField<decimal> m_Saldo = new DBCursorField<decimal>("Saldo");
            private DBCursorField<decimal> m_SaldoDisponibile = new DBCursorField<decimal>("SaldoDisponibile");
            private DBCursorField<StatoContoCorrente> m_StatoContoCorrente = new DBCursorField<StatoContoCorrente>("StatoContoCorrente");
            private DBCursorField<ContoCorrenteFlags> m_Flags = new DBCursorField<ContoCorrenteFlags>("Flags");

            /// <summary>
            /// Costruttore
            /// </summary>
            public ContoCorrenteCursor()
            {
            }

            /// <summary>
            /// Campo Nome
            /// </summary>
            public DBCursorStringField Nome
            {
                get
                {
                    return m_Nome;
                }
            }

            /// <summary>
            /// Campo Numero
            /// </summary>
            public DBCursorStringField Numero
            {
                get
                {
                    return m_Numero;
                }
            }

            /// <summary>
            /// Campo IBAN
            /// </summary>
            public DBCursorStringField IBAN
            {
                get
                {
                    return m_IBAN;
                }
            }

            /// <summary>
            /// Campo SWIFT
            /// </summary>
            public DBCursorStringField SWIFT
            {
                get
                {
                    return m_SWIFT;
                }
            }

            /// <summary>
            /// Campo IDBanca
            /// </summary>
            public DBCursorField<int> IDBanca
            {
                get
                {
                    return m_IDBanca;
                }
            }

            /// <summary>
            /// Campo NomeBanca
            /// </summary>
            public DBCursorStringField NomeBanca
            {
                get
                {
                    return m_NomeBanca;
                }
            }

            /// <summary>
            /// Campo DataApertura
            /// </summary>
            public DBCursorField<DateTime> DataApertura
            {
                get
                {
                    return m_DataApertura;
                }
            }

            /// <summary>
            /// Campo DataChiusura
            /// </summary>
            public DBCursorField<DateTime> DataChiusura
            {
                get
                {
                    return m_DataChiusura;
                }
            }

            /// <summary>
            /// Campo Saldo
            /// </summary>
            public DBCursorField<decimal> Saldo
            {
                get
                {
                    return m_Saldo;
                }
            }

            /// <summary>
            /// Campo SaldoDisponibile
            /// </summary>
            public DBCursorField<decimal> SaldoDisponibile
            {
                get
                {
                    return m_SaldoDisponibile;
                }
            }

            /// <summary>
            /// Campo StatoContoCorrente
            /// </summary>
            public DBCursorField<StatoContoCorrente> StatoContoCorrente
            {
                get
                {
                    return m_StatoContoCorrente;
                }
            }

            /// <summary>
            /// Campo Flags
            /// </summary>
            public DBCursorField<ContoCorrenteFlags> Flags
            {
                get
                {
                    return m_Flags;
                }
            }

            //protected internal override CDBConnection GetConnection()
            //{
            //    return APPConn;
            //}

            //protected override Sistema.CModule GetModule()
            //{
            //    return ContiCorrente.Module;
            //}

            //public override string GetTableName()
            //{
            //    return "tbl_ContiCorrenti";
            //}
        }
    }
}